cpp/src/arrow/compute/compute-test.cc

```diff
@@ -670,6 +671,15 @@ TEST_F(TestCast, FromNull) {
   ASSERT_ARRAYS_EQUAL(*result, *result);
 }
 
+TEST_F(TestCast, EmptyArray) {
```

Comment:
I'm not sure why it's in `TestCast`, since it doesn't seem to have anything to do with casting?